trie/hashtrie.go

```diff
@@ -121,3 +124,173 @@ func (t *HashTrie) Hash() common.Hash {
 	t.root = cached
 	return common.BytesToHash(hashed.(hashNode))
 }
+
+type StackTrieItem struct {
+	ext          shortNode
+	branch       fullNode
```

Comment:
Random thought: wouldn't it be simpler to _not_ use `shortNode`/`fullNode`:s, but instead just have a list of children, and extension? 
At the point of "done with this part" we'd turn the thing into whatever it needs to be -- a shortnode or a fullnode and then hash it. I'm just thinking that maybe a custom abstraction is easier to work with than the actual short/full nodes.. ?

Reply:
I see several ways of interpreting this, let's discuss this tomorrow after standup.